lib/spack/spack/cmd/__init__.py

```diff
@@ -177,6 +178,33 @@ def parse_specs(
     if not concretize:
         return specs
 
+    # Special case for concretizing a single spec
+    if len(specs) == 1:
+        return [specs[0].concretized()]
+
+    # Special case if every spec has an abstract spec
+    if all(spec.concrete or spec.abstract_hash for spec in specs):
+        ret = [s.concretized() for s in specs]
```

Comment:
```suggestion
        ret = [s.lookup_hash() for s in specs]
```
isn't lookup enough here?

Reply:
yep